GameSalesApi/GameSalesApi/Features/Authorization/AuthController.cs

```diff
@@ -0,0 +1,83 @@
+﻿using System;
+using Infrastructure.CommandBase;
+using Infrastructure.InfrastructureCommandDecorators;
+using Infrastructure.InfrastructureQueryDecorators;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Infrastructure.Result;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using DataAccess;
+using Microsoft.AspNetCore.Authorization;
+using Infrastructure.HandlerBase;
+
+namespace GameSalesApi.Features.Authorization
+{
+    [Route("api/[controller]")]
+    [AllowAnonymous]
+    [ApiController]
+    public class AuthController : ControllerBase
+    {
+        private readonly GameSalesContext _dbContext;
+        private readonly ILogger<AuthController> _logger;
+        private readonly TokenCreator _tokenCreator;
+
+        public AuthController(GameSalesContext dbContext, ILogger<AuthController> logger, TokenCreator tokenCreator)
+        {
+            _dbContext = dbContext;
+            _logger = logger;
+            _tokenCreator = tokenCreator;
+        }
+
+        [HttpPost("login")]
+        public IActionResult Login([FromBody] LoginCommand loginCommand)
+        {
+            var builder = new CommandDecoratorBuilder<LoginCommand, Result<TokenDTO>>();
+
+            var handler = builder
+                .Add<LoginCommandHandler>()
+                    .AddParameter<GameSalesContext>(_dbContext)
+                    .AddParameter<TokenCreator>(_tokenCreator)
+                .Add<ValidationCommandDecorator<LoginCommand, Result<TokenDTO>>>()
+                .Add<LoggerCommandDecorator<LoginCommand, Result<TokenDTO>>>()
+                    .AddParameter<ILogger<AuthController>>(_logger)
+                .Add<ProfilerCommandDecorator<LoginCommand, Result<TokenDTO>>>()
+                .Add<SaveChangesCommandDecorator<LoginCommand, Result<TokenDTO>>>()
+                    .AddParameter<ICommandDispatcher>(null)
+                    .AddParameter<GameSalesContext>(_dbContext)
+                .Build();
```

Comment:
Is it possible to move the same parts of decorator creating to a separate class? (extensions methods)
Or create them by DI?

Reply:
I thought it will be builder.AddBase(a, b, c).AddNeedableComment().Build()